### ISSUE TITLE
ValidFunctionName: ignore deprecated methods by design

### ISSUE THREAD

**Issue by jrfnl** [WordPress/WordPress-Coding-Standards#1797]
## Is your feature request related to a problem?

Functions and methods should comply with the WP Coding Standards of being `snake_case`.

When a (`public`/`protected`) non-snake-case method/function is renamed, however, it breaks backward-compatibility, so it is common to deprecate the old method name and add a new method with a name which complies to the coding standards.

## Describe the solution you'd like
I'm proposing that if a function/method is marked as `deprecated` in the function docblock, that the sniff should not throw an error.

```php
class Example {
    public function new_name() {
    }

    /**
     * @deprecated x.x.x Use Example::new_name() instead.
     */
    public function oldName() {
        return self::new_name();
    }
}
```

Currently the sniff would throw an error for the `oldName()` method. Once this enhancement has been implemented, it no longer should.
